xlthlx/news-widget-block
# Add News block plugin

## Description

This PR adds the logic for the News Widget Block.
It uses `@wordpress/env` to set up a local environment and `@wordpress/create-block` to create the structure of a new block.
The implementation for this block is very basic: it creates a Settings page where to insert the API key, it retrieves the contents from the API URL and shows them in the frontend and in the backend using a callback rendering of the block itself.

## How to test

Install the dependencies:

 `npm install`

Start the local environment with:

  `npx wp-env start`
  
The local environment will be available at the address http://localhost:1188

To use the block, you should add your News API key first in the [Settings](http://localhost:1188/wp-admin/admin.php?page=news-widget-block-settings).

Then create a post and add the block `US News`.

To stop the local environment after the test:

  `npx wp-env stop`
